### DISCUSSION TITLE
Build with GCC MinGw-64

### DISCUSSION THREAD

**Discussion by Axle-Ozz-i-sofT** [frang75/nappgui_src#111] [General]
Hi
I have been attempting to build the library with "winlibs-x86_64-posix-seh-gcc-13.2.0-mingw-w64msvcrt-11.0.1-r5".
I don't use system paths and set them from the command line when required.
I have attempted to build with variations of the following batch file:
```
SET PATH=%PATH;C:\Dev-Cpp-Emb_Winlibs\TDM-GCC-64\bin;C:\Dev-Cpp-Emb_Winlibs\nappgui_src-1.4.0
cmake -DCMAKE_C_COMPILER=C:/Dev-Cpp-Emb_Winlibs/TDM-GCC-64/bin/gcc.exe -DCMAKE_CXX_COMPILER=C:/Dev-Cpp-Emb_Winlibs/TDM-GCC-64/bin/g++.exe -S C:\Dev-Cpp-Emb_Winlibs\nappgui_src-1.4.0 -DBUILD_SHARED_LIBS=ON -B ./build
cmake --build ./build
```
A. It always defaults to A vcxproj MSVC compiler static.
B. I can't work out how to invoke the "Release" "Shared"?

Do you have a basic CMake command line for GCC MinGW-w64 build release, shared?
Can I use:
```
cmake -G "MinGW Makefiles" -DCMAKE_C_COMPILER=[path] -DCMAKE_CXX_COMPILER=[path] -S src path] [FLAGS release, shared] ./build
REM and then invoke make
mingw32-make
```

I am only just beginning to use CMake and a bit green working this out (More used to Make, mingw32-make).
Best regards
Axle

**Comment by frang75** [comment]
Hi @Axle-Ozz-i-sofT!
MinGw still is not supported by NAppGUI build system.
More info in this issue: https://github.com/frang75/nappgui_src/issues/48

At the moment you have to use VisualStudio for build NAppGUI in Windows.

**Reply by Axle-Ozz-i-sofT** [reply]
I compiled the library "shared" using MSVC compiler and then attempted to compile the hello world projects with GCC.
After all of the errors and looking at the source everything seams tied to MSVC and Visual Studio :(

Seeing Linux GCC was supported I automatically assumed GCC would be supported on Windows as well.
There's too much C++ code in this C library for me to work around that. I typically just work with C GCC Windows and Linux, so I don't think I can do anything with this.

Thanks for your assist and sorry to take up your time
Axle

**Comment by Axle-Ozz-i-sofT** [comment]
Ok, thank you.
I'm not strong with creating build systems CMake or make, but I should be able to compile each dll individually.
I had a look at https://github.com/frang75/nappgui_src/issues/5 (Sorry, I didn't see to separate library bins and src repos at first.)
I can't see any hints as to why I can't compile this with GCC (MinGw) so I will give it a shot when I have time.
I'll post back my success or failure :)

Thanks
Axle

P.S. What switch do I give to CMake to build release and shared with MSVC?

**Reply by frang75** [reply]
In this page you have the instructions to build NAppGUI: https://nappgui.com/en/guide/build.html

**Comment by frang75** [comment]
The support for MinGW has been added in NAppGUI 1.4.2.
Doc: https://nappgui.com/en/guide/win_mac_linux.html#h4
Commit: https://github.com/frang75/nappgui_src/commit/5f638074231f028a5eb8a07c2f0d232edad408c6
